packages/react-dom/src/__tests__/ReactCompositeComponent-test.js

```diff
@@ -225,6 +225,86 @@ describe('ReactCompositeComponent', () => {
     expect(inputProps.prop).not.toBeDefined();
   });
 
+  it('should warn about `forceUpdate` on not-yet-mounted components', () => {
+    class MyComponent extends React.Component {
+      constructor(props) {
+        super(props);
+        this.forceUpdate();
+      }
+      render() {
+        return <div />;
+      }
+    }
+
+    const container = document.createElement('div');
+    expect(() => ReactDOM.render(<MyComponent />, container)).toWarnDev(
+      "Warning: Can't call forceUpdate on a component that is not yet mounted. " +
+        'This is a no-op, but it might indicate a bug in your application. ' +
+        'To fix, assign the initial state in the MyComponent constructor. ' +
+        'If the state needs to reflect an external data source, ' +
+        'you may also add a componentDidMount lifecycle hook to MyComponent ' +
+        'and call setState there if the external data has changed.',
```

Comment:
Even though this one is for `forceUpdate` I figured we should push people to use `this.state = ...` and `setState` here anyway.

Reply:
Yeah, no strong feelings about this I guess.